mogc_partnerships/serializers.py

```diff
@@ -89,6 +91,36 @@ def get_is_enrolled(self, obj):
         return obj.offering_id in context["enrollments"]
 
 
+class CohortMembershipListSerializer(serializers.ListSerializer):
+    def create(self, validated_data):
+        cohort = self.context.get("cohort")
+        member_emails = [o["email"] for o in validated_data]
+
+        User = get_user_model()
+        membership_accounts = User.objects.filter(
+            email__in=[email for email in member_emails]
+        ).values_list()
+        account_email_map = {user.email: user for user in membership_accounts}
+
+        cohort_memberships = [
+            models.CohortMembership(
+                user=account_email_map.get(member_email),
+                cohort=cohort,
+                email=member_email,
+            )
+            for member_email in member_emails
+        ]
+
+        objects = models.CohortMembership.objects.bulk_create(
+            cohort_memberships, ignore_conflicts=True
+        )
+        # bulk_create doesn't return autoincremented IDs with MySQL DBs
+        # so we have to query results separately
+        return models.CohortMembership.objects.filter(
+            email__in=[cm.email for cm in objects]
+        ).all()
```

Comment:
Should this also filter the QuerySet to only include memberships associated with `cohort`?

Also, I think the `.all()` is not necessary.